runtime/amd64.S

```diff
@@ -1119,7 +1119,7 @@ LBL(112):
 #endif
         movq    %rax, C_ARG_1
         LEA_VAR(caml_raise_unhandled_effect, %rax)
-        jmp     LBL(caml_c_call)
+        jmp     GCALL(caml_c_call)
```

Comment:
I'm worried this would break TSan, since a `TSAN_ENTER_FUNCTION` was executed just above and entering at the beginning of `caml_c_call` will result in a second call to `__tsan_enter_function`. This would create an imbalance in entry/exit, so an incorrect backtrace or a crash/assertion in TSan.

I agree, it looks like this code is missing an `ENTER_FUNCTION` since `caml_c_call` does a `LEAVE_FUNCTION`, and the `ENTER_FUNCTION` is skipped when jumping to `LBL(caml_c_call)`.

Reply:
You're right, this is wrong. I'll keep amd64 unchanged for the time being (i.e. I'll revert that).

Reply:
> This would create an imbalance in entry/exit, so an incorrect backtrace or a crash/assertion in TSan.

We're calling a function that raises an exception.  Doesn't this always create "an imbalance in entry/exit" ?

Reply:
We have some subroutines in the runtime, which are called from `caml_raise_exn` or `caml_raise_exception`, that scan the stack from an exception raising point up to the first effect handler and call `__tsan_func_exit` to balance entry/exit for TSan.

Imbalance with frame-pointers may be fine since `rbp` is restored at `Lentertrap` but we need to be sure that previous function epilogues don't mess with the stack.